server/src/main/java/org/elasticsearch/common/regex/Regex.java

```diff
@@ -113,8 +113,12 @@ public static boolean simpleMatch(String pattern, String str) {
             }
             return false;
         }
+        final String patternBeforeFirstIndex = pattern.substring(0, firstIndex);
+        if (firstIndex == pattern.length() - 1) {
+            return str.startsWith(patternBeforeFirstIndex);
+        }
         return (str.length() >= firstIndex &&
-                pattern.substring(0, firstIndex).equals(str.substring(0, firstIndex)) &&
+                patternBeforeFirstIndex.equals(str.substring(0, firstIndex)) &&
```

Comment:
While we are micro optimizing :), why not leave this as is and also inline the `pattern.substring(0, firstIndex)` inside the new conditional above? That way we still get the free breakout for short strings in the first part of this boolean?